src/utils/relayer.ts

```diff
@@ -0,0 +1,107 @@
+import { Redis } from 'ioredis'
+import { Node } from '@pokt-network/pocket-js'
+import { SyncCheckOptions } from '../services/sync-checker'
+import { BlockchainsRepository } from '../repositories'
+import { BlockchainDetails } from './types'
+
+// Fetch node client type if Ethereum based
+export async function fetchClientTypeLog(
+  redis: Redis,
+  blockchainID: string,
+  id: string | undefined
+): Promise<string | null> {
+  const clientTypeLog = await redis.get(blockchainID + '-' + id + '-clientType')
+
+  return clientTypeLog
+}
+
+export function isCheckPromiseResolved(promise: PromiseSettledResult<Node[]>): boolean {
+  return promise.status === 'fulfilled' && promise.value !== undefined && promise.value.length > 0
+}
+
+export function filterCheckedNodes(syncCheckNodes: Node[], chainCheckedNodes: Node[]): Node[] {
+  // Filters out nodes that passed both checks.
+  const nodes = syncCheckNodes.filter((syncCheckNode) =>
+    chainCheckedNodes.some((chainCheckedNode) => syncCheckNode.publicKey === chainCheckedNode.publicKey)
+  )
+
+  return nodes
+}
+
+// Load requested blockchain by parsing the URL
+export async function loadBlockchain(
+  host: string,
+  redis: Redis,
+  blockchainsRepository: BlockchainsRepository,
+  defaultLogLimitBlocks: number
+): Promise<BlockchainDetails> {
+  // Load the requested blockchain
+  const cachedBlockchains = await redis.get('blockchains')
+  let blockchains
+
+  if (!cachedBlockchains) {
+    blockchains = await blockchainsRepository.find()
+    await redis.set('blockchains', JSON.stringify(blockchains), 'EX', 60)
+  } else {
+    blockchains = JSON.parse(cachedBlockchains)
+  }
+
+  // Split off the first part of the request's host and check for matches
+  const [blockchainRequest] = host.split('.')
+
+  const blockchainFilter = blockchains.filter(
```

Comment:
Seeing that we're accessing the first array element only, we could probably do
```suggestion
  const [blockchainFilter] = blockchains.filter(
```

And that way we can just check below if it's defined (or if it's not and return, like I had suggested above in how to treat `if`s).